pkg/helpers/cmd/errors.go

```diff
@@ -30,3 +35,75 @@ func CheckPodSecurityErr(err error) {
 
 	kcmdutil.CheckErr(err)
 }
+
+// CheckOAuthDisabledErr checks to see if originalError stems from the fact that
+// the OAuth server is disabled.
+func CheckOAuthDisabledErr(originalError error, discoveryClient discovery.DiscoveryInterface, groups ...string) {
+	if agg, ok := originalError.(utilerrors.Aggregate); ok && len(agg.Errors()) == 1 {
+		originalError = agg.Errors()[0]
+	}
+
+	if originalError == nil {
+		return
+	}
+
+	if discoveryClient == nil {
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+
+	// only NotFound errors can stem from missing GRs, so short-circuit if we're
+	// dealing with any other error
+	if !apierrors.IsNotFound(originalError) {
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+
+	status, ok := originalError.(apierrors.APIStatus)
+	if !ok && !errors.As(originalError, &status) {
+		// this should never happen, since `apierrors.IsNotFound()` succeeded above
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+	details := status.Status().Details
```

Comment:
if ok==false, why is this line accurate?

Reply:
Either `ok` is true, or `errors.As` returns true, in both cases `status` is a non-nil `apierrors.APIStatus`. If both fail, we exit on L66.